src/c++/library/common.h

```diff
@@ -530,6 +530,10 @@ class InferResult {
   /// \return Error object indicating the success or failure.
   virtual Error IsNullResponse(bool* is_null_response) const = 0;
 
+  /// Get output for this response.
+  /// \return Error object indicating the success or failure.
+  virtual Error Output(std::string& output) const = 0;
```

Comment:
@tgerdesnv I decided to make a dedicated new API just for getting the "output" of an infer result. The existing `RawData()` required an output name be specified, which I didn't want the user of this API (PA developer) to have to care about. The OpenAI Client Backend can implement this and grab the content field of the output message.

Reply:
One concern I have about the new API: Does it also work with response outputs other than just strings (e.g. images)? The `RawData()` seems generic enough to work with any kind of output types.

Reply:
I still feel like this is the wrong approach. Adding a single string output into the InferResult interface that is used for every backend/model feels too dirty, even for a quick and dirty implementation. We are adding a function to an interface that is only used in one specific case, and won't even handle other cases we may want moving forward. Every client/backend is going to need to have more knowledge than it should to be able to grab specific parts of data and put them in specific places.

I agree that the PA developer shouldn't need to know this task-specific information. Why can't we just stick with RawData and keep ALL of the outputs, and have the task-specific script (GenAI-PA) figure out what it needs? In the places that you are calling Output, the very next line is ValidateOutputs which shows how to iterate through all the outputs for the model and grab the data.

Reply:
Agreed with Tim: https://github.com/triton-inference-server/client/pull/478#discussion_r1508066566

Reply:
Dumping all outputs now